src/frontend/Ast_to_Mir.ml

```diff
@@ -1,44 +1,54 @@
 open Core_kernel
+open Common
 open Middle
 
 (* XXX fix exn *)
 let unwrap_return_exn = function
-  | Some (ReturnType ut) -> ut
-  | x -> raise_s [%message "Unexpected return type " (x : returntype option)]
+  | Some (UnsizedType.ReturnType ut) -> ut
+  | x ->
+      raise_s
+        [%message
+          "Unexpected return type " (x : UnsizedType.returntype option)]
 
 let trans_fn_kind = function
-  | Ast.StanLib -> StanLib
+  | Ast.StanLib -> Fun_kind.StanLib
   | UserDefined -> UserDefined
 
 let rec op_to_funapp op args =
   let argtypes =
     List.map ~f:(fun x -> (x.Ast.emeta.Ast.ad_level, x.emeta.type_)) args
   in
-  { expr= FunApp (StanLib, string_of_operator op, trans_exprs args)
-  ; emeta=
-      { mtype= operator_return_type op argtypes |> unwrap_return_exn
-      ; mloc= Ast.expr_loc_lub args
-      ; madlevel= Ast.expr_ad_lub args } }
+  let type_ =
+    Stan_math_signatures.operator_stan_math_return_type op argtypes
+    |> unwrap_return_exn
+  and loc = Ast.expr_loc_lub args
+  and adlevel = Ast.expr_ad_lub args in
+  Expr.
+    { Fixed.pattern= FunApp (StanLib, Operator.to_string op, trans_exprs args)
+    ; meta= Expr.Typed.Meta.create ~type_ ~adlevel ~loc () }
 
 and trans_expr {Ast.expr; Ast.emeta} =
-  let mtype = emeta.Ast.type_
-  and mloc = emeta.loc
-  and madlevel = emeta.ad_level in
+  let type_ = emeta.Ast.type_
+  and loc = emeta.loc
+  and adlevel = emeta.ad_level in
   match expr with
   | Ast.Paren x -> trans_expr x
   | BinOp (lhs, And, rhs) ->
-      { expr= EAnd (trans_expr lhs, trans_expr rhs)
-      ; emeta= {madlevel; mloc; mtype} }
+      Expr.
+        { Fixed.pattern= EAnd (trans_expr lhs, trans_expr rhs)
+        ; meta= Typed.Meta.create ~type_ ~adlevel ~loc () }
```

Comment:
Why not use the record syntax here?

Reply:
Again, I think hiding implementation and exposing helpers for constructing expressions and statements would make the code a lot easier to read.